Sources/LeafKit/LeafAST.swift

```diff
@@ -0,0 +1,128 @@
+public typealias ResolvedDocument = LeafAST
+
+// Internal combination of rawAST = nil indicates no external references
+// rawAST non-nil & flat = false : unresolved with external references, and public AST is not flat
+// rawAST non-nil & flat = true : resolved with external references, and public AST is flat
+public struct LeafAST: Hashable {
+    public func hash(into hasher: inout Hasher) { hasher.combine(name) }
```

Comment:
It's treated that way by the pipeline - this was one of my struggles when looking through originally. "template", "path", "name" are all basically analogous -

render("aSpecificTemplate") is using "aSpecificTemplate" as an identifier throughout, though it previously was (unwisely I think) using the expanded "/root/path/to/aSpecificTemplate.leaf" to cache the document on.

Reply:
It's understandable to use the path, considering that two templates with the same filename could easily exist in different nearby directories. At the same time, filename in general is a questionable cache key, given that ideally Leaf should be able to function as an abstract parser which doesn't require an explicit filesystem at all (in other words you should be able to evaluate inline strings as templates, and I _don't_ mean by providing a screwball `LeafFiles` implementation). Ironically, this is similar to issues encountered by both `swiftc` and `clang` (mostly thanks to `import` and `#include` (respectively), though there are other implicit filesystem bindings involved as well). Even languages like PHP have that problem. Lua is the only one I can think of offhand whose interpreter doesn't have to do any monkey business whatsoever to exist in a completely filesystem-less universe.

Reply:
Sure but meaning the EXPANDED path is used as the key is pointless in the current implementation; more specifically, referencing
`render("this/template")` and `render("that/template")`, they're both path/name/template as "this/template" and "that/template" - the only other instance is using the FULL path, and since the only template source is file store, they'd both have the same information added to them.